### PR TITLE
Print failed build commands in better format for debugging and testing

### PR DESCRIPTION
**Pull request by AlexBrownAMD**

### CODE REVIEW

File: Tensile/KernelWriter.py
Comment (nakajee) [inline Tensile/KernelWriter.py:5102]: I am not familiar with error handling code.
1) Why we need to check the type of exc.cmd? Can it be a different type?
2) How the output message changed (before and after)?
3) It looks like exc also has output (or stdout, stderr). Don't we need to print it as well?

Comment (AlexBrownAMD) [reply Tensile/KernelWriter.py:5102]: subprocess accepts arguments as either a string or a sequence, most places in our code pass a list, but the check ensures both cases are treated correctly.
When the command is a list, the default stringification for the exception prints it to look like a python list like
Command ['cmd', '-arg1', 'val1', ...]
To make it easier to debug, this change reformats the list to the actual command-line string so you can easily copy-paste and re-run the failed command by itself.
Printing stdout+stderr is already taken care of elsewhere in the code.

Comment (nakajee) [reply Tensile/KernelWriter.py:5102]: Thanks for your quick response.
I do not have any further comment.
Waiting for comments(or approval) from others.

**Comment by nakajee** [comment]
Is the preCheckin fail due to your change?

**Comment by AlexBrownAMD** [comment]
> Is the preCheckin fail due to your change?

I don't think so. Looks like it timed out on 908, which has been happening sporadically on CI for a bit now

**Comment by nakajee** [comment]
OK. It might be better to re-run pre_checkin tests to make sure this is not caused by your change.
I did not see similar failure in other PR.

**Comment by nakajee** [comment]
> OK. It might be better to re-run pre_checkin tests to make sure this is not caused by your change. I did not see similar failure in other PR.

The pre_checkin test seems OK this time.
I do not have any further comment.

I would like to have some comment/approval from other reviewers.

**Review by nakajee — APPROVED** [review]

**Pull request merged**
